sdk/storage/Azure.Storage.Common/src/Shared/LazyLoadingReadOnlyStream.cs

```diff
@@ -136,13 +142,14 @@ public async Task<int> ReadInternal(byte[] buffer, int offset, int count, bool a
 
             }
 
-            if (_bufferPosition == 0 || _bufferPosition == _bufferLength)
+            if (_bufferPosition == 0 || _bufferPosition == _bufferLength || _seeked)
             {
                 int lastDownloadedBytes = await DownloadInternal(async, cancellationToken).ConfigureAwait(false);
                 if (lastDownloadedBytes == 0)
                 {
                     return 0;
                 }
+                _seeked = false;
```

Comment:
In the code 10 lines above this, do you need to change all the condition checks from `_position == _length` to `_position >= _length`?  What happens in the case where I allow modifications, seek past the end of the blob, and then try to read again?  If we update the length above we'll return 0 bytes read.  If you don't update the  length, you'll drop into `DownloadInternal` and that'll ask for a range beyond the end of the blob which will cause an exception.

We should also add more test cases for `allowBlobModifications`:
- Seek Beyond End and Find Nothing
    ```C#
    stream.Seek(stream.Length + 100);
    byte[] data = new byte[1];
    Assert.Zero(stream.Read(data, 0, 1));
    ```
- Seek Beyond End and Find Data
    ```C#
    stream.Seek(stream.Length + 100);
    underlyingBlob.Upload(longerDocument);
    byte[] data = new byte[1];
    Assert.NotZero(stream.Read(data, 0, 1));
    ```

Reply:
fixed.

Reply:
Did you fix this?  The code on line 120/126 is still just checking `if (_position == _length)`.  I'm talking about this feedback in particular:

> In the code 10 lines above this, do you need to change all the condition checks from _position == _length to _position >= _length? What happens in the case where I allow modifications, seek past the end of the blob, and then try to read again? If we update the length above we'll return 0 bytes read. If you don't update the length, you'll drop into DownloadInternal and that'll ask for a range beyond the end of the blob which will cause an exception.

Unless I was mistaken about the need for a fix here because the extra unit tests showed no problems?

Reply:
The new tests pass as expected.  I get 0 bytes read if there is no data, and the data I was expecting when it exists.

Reply:
Hmm.  I was worried about a 416 per https://docs.microsoft.com/en-us/rest/api/storageservices/specifying-the-range-header-for-blob-service-operations#format-1-bytesstartbyte- if we sent an invalid range.

Reply:
There's a bug in `OpenReadAsync_Seek_NewPositionGreaterThanBlobLength_AllowBlobModificationsTrue_NoData` test. After fixing it's going to detect it (I left a comment next to it).